nixos/tests/mongodb.nix

```diff
@@ -12,29 +12,76 @@ in {
   };
 
   nodes = {
-    one =
-      { ... }:
-        {
-          services = {
-           mongodb.enable = true;
-           mongodb.enableAuth = true;
-           mongodb.initialRootPassword = "root";
-           mongodb.initialScript = pkgs.writeText "mongodb_initial.js" ''
-             db = db.getSiblingDB("nixtest");
-             db.createUser({user:"nixtest",pwd:"nixtest",roles:[{role:"readWrite",db:"nixtest"}]});
-           '';
-           mongodb.extraConfig = ''
-             # Allow starting engine with only a small virtual disk
-             storage.journal.enabled: false
-             storage.mmapv1.smallFiles: true
-           '';
-          };
-        };
+    v3_4 = { ... }:
+    {
+      services = {
+        mongodb.enable = true;
+        mongodb.package = pkgs.mongodb-3_4;
+        mongodb.enableAuth = true;
+        mongodb.initialRootPassword = "root";
+        mongodb.initialScript = pkgs.writeText "mongodb_initial.js" ''
+            db = db.getSiblingDB("nixtest");
+            db.createUser({user:"nixtest",pwd:"nixtest",roles:[{role:"readWrite",db:"nixtest"}]});
+        '';
+        mongodb.extraConfig = ''
+            # Allow starting engine with only a small virtual disk
+            storage.journal.enabled: false
+            storage.mmapv1.smallFiles: true
+        '';
+      };
     };
 
+    v3_6 = { ... }:
+    {
+      services = {
+        mongodb.enable = true;
+        mongodb.package = pkgs.mongodb-3_6;
+        mongodb.enableAuth = true;
+        mongodb.initialRootPassword = "root";
+        mongodb.initialScript = pkgs.writeText "mongodb_initial.js" ''
+              db = db.getSiblingDB("nixtest");
+              db.createUser({user:"nixtest",pwd:"nixtest",roles:[{role:"readWrite",db:"nixtest"}]});
+        '';
+        mongodb.extraConfig = ''
+              storage.journal.enabled: false
+        '';
+      };
+    };
+
+    v4_0 = { ... }:
+    {
+      services = {
+        mongodb.enable = true;
+        mongodb.package = pkgs.mongodb-4_0;
+        mongodb.enableAuth = true;
+        mongodb.initialRootPassword = "root";
+        mongodb.initialScript = pkgs.writeText "mongodb_initial.js" ''
+              db = db.getSiblingDB("nixtest");
+              db.createUser({user:"nixtest",pwd:"nixtest",roles:[{role:"readWrite",db:"nixtest"}]});
+        '';
+        mongodb.extraConfig = ''
+              storage.journal.enabled: false
+        '';
+      };
+    };
+
+
+  };
+
   testScript = ''
-    startAll;
-    $one->waitForUnit("mongodb.service");
-    $one->succeed("mongo -u nixtest -p nixtest nixtest ${testQuery}") =~ /hello/ or die;
+         $v3_4->start;
```

Comment:
I think it would be better to have generic testScript, that's version independent.